BranchMetrics/ios-branch-deep-linking-attribution
# Sdk 1480 - UIPasteControl Support

## Reference
SDK-1480 - UIPasteControl iOS 16 - Allow customers to use their own UIPasteControl and pass params to Branch SDK

## Summary
* Added API passPasteItemProviders
* Added class BranchPasteControl
* Updated class BranchOpenRequest and BranchInstallRequest to send localURL if present.
* Added a new view in Testbed App which has BranchPasteControl and UIPasteControl buttons.
* Added unit test for API passPasteItemProviders

## Motivation 

Provide support for iOS 16 new control - UIPasteControl

 ` // Pass pasteboard data
 [[Branch getInstance] passPasteItemProviders:itemProviders];`

Added class BranchPasteControl - wrapper over UIPasteControl

`CGRect rectBC = CGRectMake(0, 0, _branchPasteControlView.frame.size.width, _branchPasteControlView.frame.size.height);
        BranchPasteControl *bc = [[BranchPasteControl alloc] initWithFrame:rectBC AndConfiguration:nil];
        [_branchPasteControlView addSubview:bc];`

## Type Of Change
- [ X] New feature (non-breaking change which adds functionality)
- [ X] This change requires a documentation update

## Testing Instructions

* Run unit test - testPassPasteControl
* Click on the link https://bnctestbed.app.link/nativelink
* Install TestBed 
* Press button "Show Paste control view" -> Press UIPasteControl or BranchPasteControl button. Verify logs.
